File: src/ch3/Main.java
package ch3;

import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        ArrayList<String> a = new ArrayList<>();
        a.add("a");
        a.add("b");
        a.add("c");
        System.out.println(StringFunctions.badJoinToStringWithDefaults(a, "| "));
        System.out.println(StringFunctions.lastChar("abc"));
        System.out.println(StringFunctions.jointToString(a, ";"));
    }
}

File: src/ch3/main.kt
package ch3

import ch3.os.parseFile

fun main(args: Array<String>) {
    val set = hashSetOf(1, 7, 3, 7)
    val list = arrayListOf(1, 7, 3, 7)
    val map = hashMapOf(1 to "one", 5 to "five", 7 to "seven", 1 to "one")
    val stringSet = hashSetOf("one", "two", "three")

    println(set)
    println(list)
    println(map)

    println(list.last())
    println(set.max())

    println(badJoinToString(list, "; ", "(", ")"))
    println(badJoinToString(set, separator = " ", postfix = "]", prefix = "["))

    println(badJoinToStringWithDefaults(set))

    println("Strings".lastChar())

    println(list.jointToString(separator = "; "))
    println(stringSet.join())

    println(multiple("a", "b", "c", "d"))
    val elems = arrayOf("A", "B", "C", "D")
    println(multiple(*elems))

    val file = parseFile("/home/omar/.oh-my-zsh/plugins/git/git.plugin.zsh")
    file.innerTest()

    println(file)
}

fun multiple(vararg elems: String): String = elems.joinToString("; ")